packages/flutter_tools/lib/src/run.dart

```diff
@@ -198,8 +197,6 @@ class RunAndStayResident extends ResidentRunner {
     }
 
     printStatus('Application running.');
-    if (debuggingOptions.buildMode == BuildMode.release)
```

Comment:
Can we keep this line and the next to preserve the `flutter run --release` behavior
by modifying the `if` expression?
```
if (isNotDaemonMode && debuggingOptions.buildMode == BuildMode.release)
```